test/ng/directive/ngEventDirsSpec.js

```diff
@@ -149,5 +149,75 @@ describe('event directives', function() {
       expect(element.val()).toBe('newValue');
     }));
 
+
+    it('should throw the error when wrapped in an apply',
+        inject(function($rootScope, $compile) {
+      element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+
+      $rootScope.click = function() {
+        throw new Error('listener error');
+      };
+
+      expect(function() {
+        $rootScope.$apply(function() {
+          element.triggerHandler('click');
+        });
+      }).toThrowError('listener error');
```

Comment:
I thought this error would be caught in the [`$apply` phase try/catch](https://github.com/Narretz/angular.js/blob/61c8910d7547a691f9d6f63129d0e16ffb4f93db/src/ng/rootScope.js#L1188) 🤔 

Also I think we could use another test which invokes the handler within a watcher, so it would be [within the `$digest` phase](https://github.com/Narretz/angular.js/blob/61c8910d7547a691f9d6f63129d0e16ffb4f93db/src/ng/rootScope.js#L849)

But note that in both of these cases it does not re-throw, so why does it in this test?

Reply:
I haven't confirmed, but I suspect it is because `$exceptionHandler` throws by default in tests. In order to test the "real" behavior, you have to set `$exceptionHandler` in `log` mode.

Reply:
Ah yes, that sounds right. I guess we should have tests for both cases?